src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with logical operators (&&, ||) or ternary operator (?:)
-  // without nesting
+  const men = people
+    .filter(item =>
+      item.sex === 'm'
+      && (century === undefined
+        || (item.died < century * 100 && item.died >= (century - 1) * 100)))
+    .map(item => item.died - item.born);
```

Comment:
Your code looks intuitive, but you iterate array 3 times, can you move all code to reduce?